airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/DiscoverOperation.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.operation
+
+import io.airbyte.cdk.command.SourceConfiguration
+import io.airbyte.cdk.consumers.OutputConsumer
+import io.airbyte.cdk.discover.ColumnMetadata
+import io.airbyte.cdk.discover.DiscoveredStream
+import io.airbyte.cdk.discover.MetadataQuerier
+import io.airbyte.cdk.discover.TableName
+import io.airbyte.protocol.models.v0.AirbyteCatalog
+import io.airbyte.protocol.models.v0.AirbyteStream
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Singleton
+
+private val log = KotlinLogging.logger {}
+
+@Singleton
+@Requires(property = CONNECTOR_OPERATION, value = "discover")
+@Requires(env = ["source"])
```

Comment:
nice - we are still using these ENV tags

Reply:
This here might not be the most relevant use case (it's probably redundant) but constraints on `env` are particularly useful in tests.